endpoints.tf

```
@@ -13,3 +13,147 @@ resource "aws_vpc_endpoint" "transfer_server" {
   vpc_endpoint_type   = "Interface"
   vpc_id              = aws_vpc.default.id
 }
+
+
+data "aws_region" "current" {}
+
+resource "aws_vpc_endpoint" "s3" {
+  count             = var.private_s3_endpoint != false ? 1 : 0
+  service_name      = "com.amazonaws.${data.aws_region.current.name}.s3"
+  tags              = merge(var.tags, { "Name" = "s3-vpc-endpoint-${var.name}" })
+  vpc_endpoint_type = "Gateway"
+  vpc_id            = aws_vpc.default.id
+  route_table_ids   = aws_route_table.private[*].id
+}
+
+data "aws_vpc_endpoint_service" "ssm_endpoint" {
+  count   = length(var.ssm_endpoint.subnet_ids) != 0 ? 1 : 0
+  service = "ssm"
+}
+
+resource "aws_security_group" "sgp_ssm_endpoint" {
```

As these are all different resource types, they can all share the same name. So we don't have to prefix them with something the represents the type of resource:

```suggestion
resource "aws_security_group" "ssm_endpoint" {
```